### PR TITLE
Move the bootloader usb descriptors to a separate file

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: Bootloader/firmware/src/bootloader_usb_descriptors.c [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  0, // Configuration string index
+  USB_ATTRIBUTE_DEFAULT | USB_ATTRIBUTE_SELF_POWERED, // Attributes
+  50, // Max power consumption (2X mA)
+
+  // DFU Interface Descriptor
+  0x09, // Size of this descriptor in bytes
+  USB_DESCRIPTOR_INTERFACE, // Descriptor Type
+  DFU_MODE_DFU_INTERFACE_INDEX, // Interface Number
+  DFU_ALT_INTERFACE_FIRMWARE, // Alternate Setting Number
+  0x00, // Number of endpoints in this intf
```
Comment (peternewman) [inline Bootloader/firmware/src/bootloader_usb_descriptors.c]: SPaG interface

File: Bootloader/firmware/src/bootloader_usb_descriptors.c [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+  0xfe, // Class code
+  0x01, // Subclass code
+  0x02, // Protocol code
+  0x04, // Interface string index
+
+  0x09, // Size of this descriptor in bytes
+  USB_DESCRIPTOR_INTERFACE, // Descriptor Type
+  DFU_MODE_DFU_INTERFACE_INDEX, // Interface Number
+  DFU_ALT_INTERFACE_UID, // Alternate Setting Number
+  0x00, // Number of endpoints in this intf
```
Comment (peternewman) [inline Bootloader/firmware/src/bootloader_usb_descriptors.c]: Again

**Comment by peternewman** [comment]
Generally LGTM

File: Bootloader/firmware/src/bootloader_usb_descriptors.c (hunk 1 above)
Comment (nomis52) [inline Bootloader/firmware/src/bootloader_usb_descriptors.c]: Done

File: Bootloader/firmware/src/bootloader_usb_descriptors.c (hunk 2 above)
Comment (nomis52) [inline Bootloader/firmware/src/bootloader_usb_descriptors.c]: Done

**Comment by peternewman** [comment]
LGTM

**Pull request merged**